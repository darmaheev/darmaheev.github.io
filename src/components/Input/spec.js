import React from "react";
import Input from "./index";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders input", () => {
  const input = shallow(<Input />);
  expect(input).toMatchSnapshot();
});
