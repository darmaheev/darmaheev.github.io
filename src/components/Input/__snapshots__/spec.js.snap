// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [
        Object {
          "type": "text",
        },
      ],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  height: var(--input-height);
  border-radius: var(--input-border-radius);
  border: 1px solid var(--input-border-color);
  padding: var(--input-padding);
  background-color: var(--input-background-color);
  color: var(--input-color);
  font-size: var(--input-font-size);
  width: var(--input-width);
  &:disabled {
    background-color: var(--button-border-color);
    opacity: 0.8;
    pointer-events: none;
  }

  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.input",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "input",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          Object {
            "type": "text",
          },
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  height: var(--input-height);
  border-radius: var(--input-border-radius);
  border: 1px solid var(--input-border-color);
  padding: var(--input-padding);
  background-color: var(--input-background-color);
  color: var(--input-color);
  font-size: var(--input-font-size);
  width: var(--input-width);
  &:disabled {
    background-color: var(--button-border-color);
    opacity: 0.8;
    pointer-events: none;
  }

  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
